### PR TITLE
api: Use source keys

### PR DESCRIPTION
**Pull request by vsreekanti**
* Change all APIs to use source keys instead of names
* Enforce uniqueness in Neo4j and Cassandra by retrieving `Item`s before creating them
* Add tests to make sure you cannot create duplicate keys

### CODE REVIEW

File: src/main/java/edu/berkeley/ground/dao/models/cassandra/CassandraEdgeFactory.java [hunk 1]
```diff
@@ -88,6 +88,20 @@ public Edge create(String name,
                      long fromNodeId,
                      long toNodeId,
                      Map<String, Tag> tags) throws GroundException {
+
+    Edge edge = null;
+    try {
+      edge = this.retrieveFromDatabase(sourceKey);
+    } catch (GroundException e) {
+      if (!e.getMessage().contains("No Edge found")) {
```
Comment (maasg) [inline src/main/java/edu/berkeley/ground/dao/models/cassandra/CassandraEdgeFactory.java:96]: Checking messages in exception is fragile and not refactor-friendly. See an alternative approach here: https://github.com/vsreekanti/ground/pull/1
